chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/RoomStore.kt

```diff
@@ -80,7 +80,10 @@ internal class RoomStore(
                 is UserSubscriptionEvent.AddedToRoomEvent ->
                     listOf(event.also { this += event.room })
                 is UserSubscriptionEvent.RoomUpdatedEvent ->
-                    listOf(event.also { this += event.room })
+                    listOf(event.also {
+                        //memberUserIds are null in this case, so append whatever we had saved last time
```

Comment:
To clarify then: the server never sends the member user IDs when it sends the room entity, they are only ever sent via a separate subscription, so anywhere we deserialise a room from the server, we will need to incorporate them.

e.g. lines 63, 73 and 81

Reply:
Okay, that seems like a thing that would have been useful to know before! Do we have any more of this information written down in regards to what information the server sends when? Or if there's no docs for that either, maybe we can set some time to go through how everything connects together and other Things that might be useful to know whilst working on the Chatkit SDK please?

And just to go back to the original point, is the comment I added now okay, or would you prefer I moved it totally?

Reply:
The API is documented in full at https://pusher.com/docs/chatkit/reference/latest, though I accept that this is not the most useful way to consume it.

The comment is fine now, though "memberUserIDs are not populated in Rooms we have just deserialised from the server because we receive them separately via membership subscriptions, so we must copy the set we have been tracking on our previous instance of the Room on to this new instance" might be better.

Reply:
Separate from the comment, we do need to take care of those other lines were we replace Room entities which we've been tracking the members with fresh-from-the-server ones which don't have any attached.

Reply:
I've updated the comment as per your suggestion. 

We don't have tests for lines 63(InitialState -> addedTo), 73(InitialState -> updated), and 81 (AddedToRoomEvent). Do you want tests that cover these cases before we push an update?

Reply:
Have added the other methods to attach the member user ids too.